manifests/client.pp

```diff
@@ -1,13 +1,13 @@
 class ssh::client(
-  $ensure               = present,
-  $storeconfigs_enabled = true,
-  $options              = {},
-  $use_augeas           = false,
-  $options_absent       = [],
+  String $ensure = present,
+  Boolean $storeconfigs_enabled = true,
+  Hash $options = {},
+  Boolean $use_augeas = false,
+  Array $options_absent = [],
 ) inherits ssh::params {
```

Comment:
@pmuller as part of renewing the module for puppet 4/5 it might also be good to get rid of inheritance in favour of modules in data pattern. All parameters are moved to a module specific hiera hierarchy which would mean a much simpler setup ultimately as it would get rid of the abundance of case statements that is currently found in params.pp